roles/aap_setup_install/tasks/main.yml

```diff
@@ -3,11 +3,11 @@
 
 - name: Check Ansible Tower Running
   uri:
-    url: "https://{{ controller_hostname }}/api/v2/ping/"
+    url: "https://{{ controller_hostname | default((aap_setup_prep_inv_nodes['automationcontroller'].keys() | list)[0]) }}"
     method: GET
-    user: "{{ controller_username }}"
-    password: "{{ controller_password }}"
-    validate_certs: "{{ controller_validate_certs | default(omit) }}"
+    user: admin
+    password: "{{ aap_setup_prep_inv_vars.all.admin_password }}"
+    validate_certs: "{{ controller_validate_certs | default('false') }}"
```

Comment:
I was default making it work out of the box for anyone, if it were normal module, I'd say yes true.

Reply:
Till now I had the impression that our approach was to do the right thing by default. Ignoring certification validation doesn't seem to me to be the "right thing" ™, or what do you think? And as our good practices state, we should define (explicit) defaults in the defaults/main.yml, not within the code!

Reply:
I'm now a little on the fence on this. If the user doesn't specify their own certs on install then this will definitely be self signed and will definitely fail. Other thing we could look at doing is adding a bit of code to trust the CA that gets created in 2.2 then we can default 'true'. Otherwise probably best to just leave as false for this because it's likely the user will either later add their own certs or will trust the certs themselves (but they've just been generated so that can't have happened yet)

Reply:
(sorry for the mind dump) 🤮

Reply:
> so we could do this with jinja and have it check if they pass a cert to default true else it will be false. I also like the idea of of adding the self signed to trust which is probably a better fix over all

Only problem I can think with that is that you could be running this from anywhere(i.e. not RHEL) I usually run it from my Mac which means that probably wont be so easy

Reply:
I thought the validate certs stuff only was for the servers connections between themselves.... is that not correct?

Reply:
This task will be likely running from localhost (or a bastion) and therefore wont be in the cluster so it wouldn't necessarily be RHEL.

In 2.2 installer a common CA is created and all the hosts trust each other which is kind of cool and we could leverage that by delegating this command to the controller host - that could potentially be better functionality anyway?

Reply:
I thought the validate certs were to set settings in controller/hub/db type stuff and that's all so it doesn't matter where this is being run from. Is that not correct?

Reply:
No, if you look this task referenced here is just calling out to the controller API to see if it's up. It would be done from the target host this role is running against (often localhost)